Orange/widgets/data/owpythonscript.py

```diff
@@ -655,8 +660,15 @@ def commit(self):
         self.console.new_prompt(sys.ps1)
         for out in self.outputs:
             signal = out.name
-            self.send(signal, self.console.locals.get(signal, None))
-
+            out_var = self.console.locals.get(signal, None)
+            if not isinstance(out_var, out.type) and out_var is not None:
+                self.Error.add_message(signal, msg=self.error_msg(signal, out.type))
```

Comment:
I'd prefer `out.type.__class__` here.

Replace "of type" with "an instance of". 

I don't see much sense in having a separate method for constructing `Msg`. I'd write

    self.Error.add_message(
        signal,
        "Variable '{}' has to be an instance of '{}'.".format(signal, out.type.__name__))